File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> less;
        vector<int> equal;
        vector<int> greater;
        for(int i=0; i<nums.size(); i++){
            if(nums[i]<pivot){
                less.push_back(nums[i]);
        }
        else if(nums[i]>pivot){
            greater.push_back(nums[i]);
        }
       else if(nums[i] == pivot){
           equal.push_back(nums[i]);
       }
        }
         less.insert(less.end(),equal.begin(),equal.end());
        less.insert(less.end(),greater.begin(),greater.end());
        
        return less;
    }
};

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
       int ans[] = new int[nums.length];
        
        
        int i = 0;
            for(int n : nums){
            if(n < pivot) ans[i++] = n;  
        }
        
        for(int n : nums){
            if(n == pivot) ans[i++] = n;  
        }
        
        for(int n : nums){
            if(n > pivot){
                ans[i++] = n;
            }
        }
        
        return ans;
    }
}